### PR TITLE
Present tool help from history entry. Issue 2100

### PR DESCRIPTION
**Pull request by tmcgowan**
Allows the user to see a tool's help text directly from the result data set entry. Added a new 
"Tool Help" button to the actions div. Clicking will toggle a div at the bottom of the data set entry. This div contains the help text found on the tool run page. PR for #2100 
help_text
new_button

### CODE REVIEW

File: client/galaxy/scripts/mvc/dataset/dataset-li.js [hunk 1]
```diff
@@ -254,6 +254,52 @@ var DatasetListItemView = _super.extend(
...
+            href: '#',
+            faIcon: 'fa-question',
+            onclick: function() {
+                var divString = 'thdiv-' + datasetID;
+                if ($("#" + divString).length > 0)
+                {
+                    $("#" + divString).toggle();     
+                } else {
+                    $.ajax({
+                    url: '/api/jobs/' + datasetID
```
Comment (dannon) [inline client/galaxy/scripts/mvc/dataset/dataset-li.js]: Should this actually be a job id, and not datasetID?

I'm getting errors like:

image

Comment (tmcgowan) [reply client/galaxy/scripts/mvc/dataset/dataset-li.js]: I'll take a look. This is my lunch diversion, so now I've got something to look forward to :)

**Comment by tmcgowan** [comment]
@dannon -  On my dev install, the creating_job id is always the same as the dataset_id for each dataset in a history, so my install was always working with the "wrong" id. Not sure why that's happening.  I clone and run.sh with no changes. Now the code explicitly refers to the creating_job id.

**Comment by guerler** [comment]
Its probably better to display the help icon as last icon in that button list. Another minor suggestion is to use a padding for the help text (5px?).

**Comment by martenson** [comment]
This feature does not render the result for me (I see no change after clicking it).

It does create two requests (`api/jobs/787898b868303ddf` and `api/tools/mergeCols1/build`) which seems one too many for this.

Also the rendering of it is wrong, it should be part of the button group:

screenshot 2016-11-15 10 08 45

**Comment by dannon** [comment]
@martenson Interesting, rendered for me.  Let me try that mergecols tool.

**Comment by martenson** [comment]
I tried on FastQC too.

**Comment by dannon** [comment]
Does select random lines work for you?

**Comment by martenson** [comment]
It does not.

**Comment by martenson** [comment]
There seems to be discrepancy with js build.

**Comment by dannon** [comment]
I'm guessing if it was throwing a client side error you'd have mentioned that by now.

**Comment by martenson** [comment]
after make client I see

```
modified:   static/scripts/bundled/analysis.bundled.js
modified:   static/scripts/bundled/analysis.bundled.js.map
modified:   static/scripts/bundled/libs.bundled.js
modified:   static/scripts/bundled/libs.bundled.js.map
modified:   static/scripts/bundled/login.bundled.js.map
```

**Comment by dannon** [comment]
Does that fix it for you, or is it a red herring?

**Comment by dannon** [comment]
(it's building cleanly for me with fresh node modules reinstalled -- can you try that?)

**Comment by martenson** [comment]
Seems like red herring, still does not work for me. Even after node_module reinstalled.

**Comment by dannon** [comment]
Ok.  Let me test a few things and try to reproduce.  I'll adjust the style there, too.

**Comment by dannon** [comment]
@martenson You weren't logged in, were you?

**Comment by martenson** [comment]
@dannon I was logged in as an admin.

**Comment by martenson** [comment]
Anonymous access seems to trigger another error (403 on the job).

**Comment by dannon** [comment]
Yep, I found that one, hence the question :)

**Comment by martenson** [comment]
I understood you found it, I wanted to have a record of it.

**Comment by dannon** [comment]
Sure.  Let me tinker with this a while, I'll open a new PR with followup fixes.

**Pull request merged**